lib/core/test/unit/Cardano/Pool/MetadataSpec.hs

```diff
@@ -83,54 +93,120 @@ import Test.QuickCheck
-    around withMockServer $ it "Mock Server works as intended" $
-        \Client{getStakePoolMetadata} -> property $ \pid -> monadicIO $ do
+spec = describe "Metadata - MockServer" $ do
+
+    around (withMockServer noCache)
+        $ it "Mock Server works as intended"
+        $ \mkClient -> property $ \pid -> monadicIO $ do
+            let Client{getStakePoolMetadata} = mkClient nullTracer
             run (getStakePoolMetadata pid) >>= \case
-                Left (FailureResponse _ res) -> do
+                Nothing -> do
                     monitor $ label "No Corresponding Metadata"
-                    monitor $ counterexample $ mconcat
-                        [ show (responseStatusCode res)
-                        , " =/= "
-                        , show status404
+                    res <- run (getStakePoolMetadata pid)
+                    monitor $ counterexample $ unwords
+                        [ "request isn't deterministic:"
+                        , show res
                         ]
-                    assert (responseStatusCode res == status404)
-                Left e -> do
-                    monitor $ counterexample $
-                        "unexpected response from server: " <> show e
-                    assert False
-                Right !metadata -> do
+                    assert (isNothing res)
+
+                Just metadata -> do
                     monitor $ label "Got Valid Metadata"
                     let json = toJSON $ ApiT metadata
                     let errs = validateJSON mempty metadataSchema json
                     monitor $ counterexample $ BL8.unpack $ Aeson.encode json
                     monitor $ counterexample $ show errs
                     assert (null errs)
 
+    around (withMockServer inMemoryCache)
+        $ it "Cache metadata when called twice within the TTL"
+        $ \mkClient -> withMaxSuccess 10 $ property $ \pid -> monadicIO $ do
+            (logs, _) <- run $ captureLogging $ \tr -> do
+                let Client{getStakePoolMetadata} = mkClient tr
+                void $ getStakePoolMetadata pid
+                void $ getStakePoolMetadata pid
+            monitor $ counterexample $ unlines $ show <$> logs
+            assert $ countMsg (_Ctor @"MsgRefreshingMetadata") logs == 1
+            assert $ countMsg (_Ctor @"MsgUsingCached")        logs == 1
+
+    around (withMockServer inMemoryCache)
+        $ it "Fetch them again when fetching outside of the TTL"
+        $ \mkClient -> withMaxSuccess 10 $ property $ \pid -> monadicIO $ do
+            (logs, _) <- run $ captureLogging $ \tr -> do
+                let Client{getStakePoolMetadata} = mkClient tr
+                void $ getStakePoolMetadata pid
+                threadDelay' (2 * defaultCacheTTL)
+                void $ getStakePoolMetadata pid
+            assert $ countMsg (_Ctor @"MsgRefreshingMetadata") logs == 2
+            assert $ countMsg (_Ctor @"MsgUsingCached")        logs == 0
```

Comment:
But now, what if the data becomes unavailable? How should it react?

Reply:
If the metadata is removed, then it'll be updated when the cache is refreshed.